clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -643,14 +643,15 @@ void CodeGenFunction::EmitOpenCLKernelMetadata(const FunctionDecl *FD,
 
   if (const IntelReqdSubGroupSizeAttr *A =
           FD->getAttr<IntelReqdSubGroupSizeAttr>()) {
-    llvm::LLVMContext &Context = getLLVMContext();
-    Optional<llvm::APSInt> ArgVal =
-        A->getValue()->getIntegerConstantExpr(FD->getASTContext());
-    assert(ArgVal.hasValue() && "Not an integer constant expression");
+
+    Expr::EvalResult Result;
+    assert(A->getValue()->EvaluateAsInt(Result, getContext()) &&
```

Comment:
`Result` will always be uninitialized when assertions are disabled. I think a better approach would be:
```
const auto *CE = dyn_cast<ConstantExpr>(A->getValue());
assert(CE && "Not an integer constant expression");
llvm::APSInt ArgVal = CE->getResultAsAPSInt();
```
This same comment applies in all the cases.

As an aside: if our CI pipeline doesn't notice the failure caused by the use of `assert` here, we may be missing some CI coverage for a release without asserts build.

Reply:
Oh thanks for catching that! I did not think of that. Will make the change now

Reply:
I just noticed you suggested `getResultAsAPSInt()`. This crashes for the same reason under discussion here - https://github.com/intel/llvm/pull/3169#issuecomment-775910725.

Reply:
Done